cli/types/cypress.d.ts

```diff
@@ -1252,14 +1252,14 @@ declare namespace Cypress {
      *
      * @see https://on.cypress.io/getcookie
      */
-    getCookie(name: string, options?: Partial<Loggable & Timeoutable>): Chainable<Cookie | null>
+    getCookie(name: string, options?: Partial<CookieOptions>): Chainable<Cookie | null>
 
     /**
      * Get all of the browser cookies.
      *
      * @see https://on.cypress.io/getcookies
      */
-    getCookies(options?: Partial<Loggable & Timeoutable>): Chainable<Cookie[]>
+    getCookies(options?: Partial<CookieOptions>): Chainable<Cookie[]>
```

Comment:
Based on conversations earlier today, do we want to hold off on allowing a domain option for getCookies() such that is only returns the current dommain & we adding a new command `cy.getAllCookies()`? So far, I'm personally in favor of using one command with options vs adding a new command of this approach but see the value in the return structure. 🤷🏻‍♀️

Reply:
We still want `cy.getCookies()` to accept a domain option, because by default it uses the AUT domain. Users need a way to specify a different domain. The domain option also gives it parity with the other cookie commands. 

I don't think `cy.getAllCookies()` will take a domain option, so it wouldn't provide the same functionality. The only overlap is `cy.getCookies({ domain: null })` will essentially be the same as `cy.getAllCookies()`. I think I might just leave `domain: null` undocumented for now, so it will be made obsolete by `cy.getAllCookies()`.

That said, we didn't really finish that discussion about this and I'm still not 100% convinced we need a `cy.getAllCookies()`, so maybe we'll hash that out some more before we start implementing it.

Reply:
+1 to hashing it out before doing the work.